src/lib/tests/symlink-manage.test.ts

```diff
@@ -0,0 +1,195 @@
+			readlink: jest.fn(),
+			mkdir: jest.fn(),
+			mount: jest.fn(),
+			isFile: jest.fn(),
+			isDir: jest.fn(),
+			unlink: jest.fn(),
+			rmdir: jest.fn(),
+			requestHandler: {
+				documentRoot: '/mock/document/root',
+			},
+		} as unknown as jest.Mocked< PHP >;
+
+		symlinkManager = new SymlinkManager( mockPHP, mockProjectPath );
+
+		jest.mock( 'path', () => ( {
+			...jest.requireActual( 'path' ),
+			basename: jest.fn( ( p: string ) => {
+				const parts = p.split( /[\\/]/ );
+				return parts[ parts.length - 1 ];
+			} ),
+			// Keep other path functions as they are
+			relative: jest.requireActual( 'path' ).relative,
+			join: jest.requireActual( 'path' ).join,
+			// Add other path functions you're using in your tests
+		} ) );
+
+		// Mock pathExists function
+		( pathExists as jest.Mock ).mockImplementation( ( fullPath: string ) => {
+			return Promise.resolve( fullPath.includes( 'existing' ) );
+		} );
+	} );
+
+	afterEach( () => {
+		jest.resetAllMocks();
+	} );
+
+	describe( 'scanAndCreateSymlinks', () => {
+		it( 'should scan for symlinks and create them in the PHP runtime', async () => {
+			const mockSymlinks = [
+				'/mock/project/path/existing/symlink1',
+				'/mock/project/path/existing/symlink2',
+			];
+
+			mockPHP.fileExists.mockImplementation( ( path ) => ! path.includes( 'vfspath' ) );
+			mockPHP.readlink.mockImplementation( ( path ) => '/mock/document/root/vfspath/' + path );
+
+			// Mock fs.readdir to return our mock symlinks
+			( fs.readdir as jest.Mock ).mockResolvedValue(
+				mockSymlinks.map( ( s ) => path.basename( s ) )
+			);
+
+			// Mock fs.lstat to indicate these are symlinks
+			( fs.lstat as jest.Mock ).mockResolvedValue( { isSymbolicLink: () => true } as Stats );
+
+			// Mock fs.realpath to return a target for each symlink
+			( fs.realpath as jest.Mock ).mockImplementation( ( p: string ) =>
+				Promise.resolve( `/real/path/to/${ path.basename( p ) }` )
+			);
+
+			await symlinkManager.scanAndCreateSymlinks();
+
+			// Verify that PHP mkdir and mount were called for each symlink
+			expect( mockPHP.mkdir ).toHaveBeenCalledTimes( 2 );
+			expect( mockPHP.mount ).toHaveBeenCalledTimes( 2 );
+
+			// Verify that the symlinks were added to the internal map
+			expect( ( symlinkManager as unknown as SymlinkManagerPrivateProperties ).symlinks.size ).toBe(
+				2
+			);
+		} );
+	} );
+
+	describe( 'startWatching and stopWatching', () => {
+		it( 'should start and stop watching for file changes', async () => {
+			const mockWatcher = {
+				[ Symbol.asyncIterator ]: jest.fn().mockImplementation( () => ( {
+					next: jest.fn().mockResolvedValue( { done: true, value: undefined } ),
+				} ) ),
+			};
+			jest.spyOn( fs, 'watch' ).mockReturnValue( mockWatcher as never );
+
+			const watchPromise = symlinkManager.startWatching();
+			expect( fs.watch ).toHaveBeenCalledWith( mockProjectPath, expect.any( Object ) );
+
+			await symlinkManager.stopWatching();
+			await watchPromise;
```

Comment:
I wonder if we could change the order here:
```suggestion
			await symlinkManager.startWatching();
			expect( fs.watch ).toHaveBeenCalledWith( mockProjectPath, expect.any( Object ) );
			await symlinkManager.stopWatching();
```